### PR TITLE
[TEST] Add TC for datarepo and tensor_trainer validation

### PR DESCRIPTION
**Pull request by songgot**
[TEST] Add TC for datarepo and tensor_trainer validation
- Add datareposrc TC
- Add datareposink TC
- Add tensor_trainer TC

nnstreamer-unittest-coverage : line 81.4%, function 94.1%

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### CODE REVIEW

File: tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+  gst_element_set_state (pipeline, GST_STATE_NULL);
+  gst_object_unref (pipeline);
+  g_main_loop_unref (loop);
+
+
+  /* Confirm file creation */
+  for (i = 0; i < 10; i++) {
+    filename = g_strdup_printf ("png_%02d.png", i);
+    file = g_file_new_for_path (filename);
+    g_file_load_contents (file, NULL, &contents, NULL, NULL, &error);
```
Comment (gichan-jang) [inline tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc:63]: Need to free contents.

File: tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+  ASSERT_NE (pipeline, nullptr);
+
+  bus = gst_pipeline_get_bus (GST_PIPELINE (pipeline));
+  gst_bus_add_watch (bus, bus_callback, loop);
+  gst_object_unref (bus);
+
+  gst_element_set_state (pipeline, GST_STATE_PLAYING);
+  g_main_loop_run (loop);
+
+  gst_element_set_state (pipeline, GST_STATE_NULL);
```
Comment (gichan-jang) [inline tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc:54]: Can we guarantee that all files are saved by datareposink when EOS occurs?
How about checking whether 10 files are arrived or not?

File: tests/nnstreamer_datarepo/datareposrc/unittest_datareposrc.cc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  g_warning ("datareposrc test");
+  try {
+    testing::InitGoogleTest (&argc, argv);
+  } catch (...) {
+    g_warning ("catch 'testing::internal::<unnamed>::ClassUniqueToAlwaysTrue'");
+  }
+
+  gst_init (&argc, &argv);  
+
+  g_warning ("here");
```
Comment (gichan-jang) [inline tests/nnstreamer_datarepo/datareposrc/unittest_datareposrc.cc:23]: Maybe this is an unnecessary message for debugging?

File: tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc (hunk 2 above)
Comment (songgot) [reply tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc]: Thank you.

File: tests/nnstreamer_datarepo/datareposrc/unittest_datareposrc.cc (hunk 3 above)
Comment (songgot) [reply tests/nnstreamer_datarepo/datareposrc/unittest_datareposrc.cc]: Thank you.

File: tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc (hunk 1 above)
Comment (songgot) [reply tests/nnstreamer_datarepo/datareposink/unittest_datareposink.cc:63]: Thank you.

File: tests/nnstreamer_datarepo/unittest_datareposrc.cc [hunk 4]
```diff
@@ -0,0 +1,782 @@
...
+  datareposrc = gst_bin_get_by_name (GST_BIN (pipeline), "datareposrc");
+  ASSERT_NE (datareposrc, nullptr);
+
+  /* set invalid param */
+  g_object_set (GST_OBJECT (datareposrc), "epochs", invalid_epochs, NULL);
+  /** value "-1" of type 'gint' is invalid or out of range for property 'start-sample-index' of type 'gint'
+     default value is set */
+  g_object_get (GST_OBJECT (datareposrc), "epochs", &get_value, NULL);
+  g_warning ("get_value: %d", get_value);
+  EXPECT_EQ (get_value, 1);
```
Comment (anyj0527) [inline tests/nnstreamer_datarepo/unittest_datareposrc.cc:652]: `EXPECT_EQ (get_value, 1U)` to avoid build warning.

Comment (songgot) [reply tests/nnstreamer_datarepo/unittest_datareposrc.cc]: Thank you.

**Review by anyj0527 — APPROVED** [review]
LGTM 👍

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**